docs/internals.md

```diff
@@ -0,0 +1,93 @@
+# How VeridicalFlow works
+
+VeridicalFlow provides three main abstractions: `vflow.vset.Vset`,
+`vflow.vfunc.Vfunc`, and `vflow.subkey.Subkey`. `Vfunc`s run arbitrary computations
+on inputs. `Vset`s collect related `Vfunc`s and determine which `Vfunc`s to
+apply to which inputs and how to do so. `Subkey`s help associate outputs with
+the `Vfunc` that produced them and the `Vset` in which the `Vfunc` was
+collected. Below we describe these three abstractions in more detail.
+
+## Vset
+
+Fundamentally, all `vflow.vset.Vset` objects generate permutations by applying
+their modules to the Cartesian product of input parameters. Internally, `Vset`
+modules and parameters are wrapped in dictionaries with tuple keys. These tuples
+contain `vflow.subkey.Subkey` objects, which identify the `vflow.vfunc.Vfunc` or
+parameter name (the `Subkey` "value"), as well as its originating `Vset` (the
+`Subkey` "origin"). Since inputs must be dictionaries with a certain format, all
+raw inputs must be initialized using `vflow.helpers.init_args`.
```

Comment:
A little confused by the description, might be helpful to show what the dictionary looks like. This is my understanding, not sure what the values in the dictionaries are:
{ (`Subkey` "value", `Subkey` "origin") : ___ }